plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryResultPageSource.java

```diff
@@ -187,6 +190,16 @@ else if (javaType == double.class) {
             else if (javaType == Slice.class) {
                 writeSlice(output, type, value);
             }
+            else if (javaType == LongTimestampWithTimeZone.class) {
+                if (type.equals(TIMESTAMP_TZ_MICROS)) {
+                    long microSeconds = (long) value;
+                    int picosOfMillis = toIntExact(microSeconds % MICROSECONDS_PER_MILLISECOND) * PICOSECONDS_PER_MICROSECOND;
+                    type.writeObject(output, fromEpochMillisAndFraction(microSeconds / MICROSECONDS_PER_MILLISECOND, picosOfMillis, UTC_KEY));
```

Comment:
instead of `/` and `%` you shouild use `floorDiv` and `floorMod`.

please add a test with values before 1970-01-01